SplitFiles/bread.c

```diff
@@ -31,12 +32,176 @@ BreadType*  rand_bread_type(BreadBin* bin) {
             }
         }
     }
+    //This should never be reached
+    printf("Error in rand_bread_type");
+    return rand_bread_type(bin);
+}
+
+BreadType* get_bread_type(BreadBin* bin, unsigned char typeLetter) {
+    if (typeLetter < 'A' || typeLetter > 'Z') {
+        return NULL;
+    }
+    int index = typeLetter - (unsigned char)'A';
+    return bin->breadTypes[index];
+}
+
+BreadState* get_bread(BreadBin* bin, unsigned char typeLetter) {
+    //int index = typeLetter - (unsigned char)'A';
```

Comment:
Create a new slice of bread, using the given letter to find the bread type.  This replaces individual functions for each bread type.